### PR TITLE
Prevent nnn from waiting on open file process

### PR DESCRIPTION
**Pull request by Rahi374**
#1 Tested and necessary on Debian.

### CODE REVIEW

**Comment by jarun** [comment]
So what happens to users on Ubuntu and OS X where the current code works?

**Comment by Rahi374** [comment]
I imagined that this version would still work, but you do have a point.
Would a build flag do the trick?

**Comment by jarun** [comment]
Yes, please add this change under a Debian-specific flag.

**Comment by jarun** [comment]
This is what I see on Ubuntu:

```
arun@CBQM882 /m/c/U/a/V/aj887757_ENHREQ1369409> cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.3 LTS"
arun@CBQM882 /m/c/U/a/V/aj887757_ENHREQ1369409> cat /etc/debian_version
stretch/sid
arun@CBQM882 /m/c/U/a/V/aj887757_ENHREQ1369409>
```

Check what's different for you on Debian. Then pass some Debian-only macro from the Makefile as explained here:

https://stackoverflow.com/a/32274389

**Comment by zmwangx** [comment]
I would guess this isn't a Debian/Ubuntu distinction, but rather one with the desktop, say Unity vs Gnome vs KDE.

**Comment by jarun** [comment]
We can keep Unity out of the equation I guess. What's the default on Debian? XFCE or Gnome?

**Comment by jarun** [comment]
OK, Gnome.

**Comment by jarun** [comment]
With the number of DEs we have and may have on Linux, this would be quite difficult to maintain and track.

I think the best way forward would be to support an env variable, which, when set, adds `F_NOWAIT`. Then we document the symptoms and ask users to set this variable if they notice this problem.

What do you guys think?

**Comment by Rahi374** [comment]
Yeah I didn't think it would be a distro problem.
I'm actually on i3 so that might affect how desktop processes are handled. I'll try it on gnome later.

I personally thought that it would be better to be decided at build time, but if it is indeed dependent on the DE then an environment variable might be better.

**Comment by jarun** [comment]
> I personally thought that it would be better to be decided at build time

As you can see, the behaviour is not defined. DEs may choose to implement it their own way and it may change even for the same distro. We can't test all of those. So the env variable will take care of that easily.

**Comment by jarun** [comment]
We also need to document this. Both under the howto section in the README and in the manpage.

What's the point of this change if users don't know it exists and how to use it? All of the effort amounts to 0 (incidentally, the default value of NNN_NOWAIT) in that case, isn't it? Or, are you thinking you are the only one affected by the issue?

File: nnn.c [hunk 1]
```diff
@@ -2863,6 +2864,9 @@ main(int argc, char *argv[])
 	/* Get the default copier, if set */
 	copier = getenv("NNN_COPIER");
 
+	/* Get nowait flag */
+	nowait = getenv("NNN_NOWAIT") ? F_NOWAIT : 0;
```
Comment (jarun) [inline nnn.c]: Why not something like:

    nowait = F_NOTRACE | (getenv("NNN_NOWAIT") ? F_NOWAIT : 0);
Didn't compile it... but something in that line. This way we can skip the extra bit-OR every time.

**Comment by jarun** [comment]
Thanks @Rahi374! I'll merge this soon.

**Comment by jarun** [comment]
@Rahi374 can you please confirm if https://github.com/jarun/nnn/commit/55643a5fcca90fe3998bee3901019fc703962083 works fine for you?

**Comment by Rahi374** [comment]
@jarun Thank you!
Your language composition is a lot better than mine.
I will leave code comments directly in the commit.

**Comment by Rahi374** [comment]
Also, do we need to increment the version number?

**Comment by jarun** [comment]
Generally we change the version number when we do a release. I'll make a release probably next month.

If you want any other features in nnn, or are interested in working as a collaborator, please let me know on gitter.

**Comment by Rahi374** [comment]
I see.

I don't usually have good ideas, plus for me this program is already perfect so I'm not sure what there is to collaborate with. Though if I do find anything I want to add I'll send you another PR

**Pull request merged**